### PR TITLE
Feature mes 5015 hcode safety component

### PR DESCRIPTION
**Pull request by rossfauldsbjss**
MES-5015 Create Highwaycode and safety question component for categories F,G,H,K (home tests) and implement state handling as well.

## Checklist

- [x] PR title includes the JIRA ticket number
- [x] Branch is rebased against the latest develop
- [x] Code has been tested manually
- [x] PR link added to JIRA ticket
- [ ] One review from each scrum team
- [ ] Squashed commit contains the JIRA ticket number

### CODE REVIEW

File: src/pages/test-report/cat-home-test/test-report.cat-home-test.page.html [hunk 1]
```diff
@@ -111,6 +111,11 @@
             <controlled-stop [testCategory]="pageState.testCategory$ | async"></controlled-stop>
           </ion-col>
         </ion-row>
+        <ion-row>
+          <ion-col>
+            <highway-code-safety [testCategory]="pageState.testCategory$ | async"></highway-code-safety>
+          </ion-col>
+        </ion-row>
```
Comment (TomBillingtonUK) [inline src/pages/test-report/cat-home-test/test-report.cat-home-test.page.html]: shouldn't this be above controlled stop

File: src/pages/test-report/components/highway-code-safety/__tests__/highway-code-safety.spec.ts [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+  DangerousFaultBadgeComponent,
+} from '../../../../../components/common/dangerous-fault-badge/dangerous-fault-badge';
+import { StartTest } from '../../../../../modules/tests/tests.actions';
+import { TestCategory } from '@dvsa/mes-test-schema/category-definitions/common/test-category';
+import { configureTestSuite } from 'ng-bullet';
+import { TestDataByCategoryProvider } from '../../../../../providers/test-data-by-category/test-data-by-category';
+import { HighwayCodeSafetyComponent } from '../highway-code-safety';
+import { HighwayCodeSafetyAddDrivingFault, HighwayCodeSafetyRemoveFault } from '../../../../../modules/tests/test-data/common/highway-code-safety/highway-code-safety.actions';
+
+describe('HighwayCodeSAfetyComponent', () => {
```
Comment (TomBillingtonUK) [inline src/pages/test-report/components/highway-code-safety/__tests__/highway-code-safety.spec.ts]: tiny typo in safety

File: src/pages/test-report/components/highway-code-safety/__tests__/highway-code-safety.spec.ts [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+        StoreModule.forRoot({ tests: testsReducer, testReport: testReportReducer }),
+      ],
+    });
+  });
+
+  beforeEach(async(() => {
+    fixture = TestBed.createComponent(HighwayCodeSafetyComponent);
+    component = fixture.componentInstance;
+    store$ = TestBed.get(Store);
+    store$.dispatch(new StartTest(105, TestCategory.B));
```
Comment (TomBillingtonUK) [inline src/pages/test-report/components/highway-code-safety/__tests__/highway-code-safety.spec.ts]: shouldn't we be starting a home test here and not B

**Review by TomBillingtonUK — APPROVED** [review]
Looks good. A couple of comments

File: src/modules/tests/test-data/common/highway-code-safety/highway-code-safety.actions.ts [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+}
+export class HighwayCodeSafetyAddSeriousFault implements Action {
+  readonly type = HIGHWAYCODE_SAFETY_ADD_SERIOUS_FAULT;
+}
+
+export class HighwayCodeSafetyRemoveFault implements Action {
+  readonly type = HIGHWAYCODE_SAFETY_REMOVE_FAULT;
+}
+
+export class AddHighwayCodeSafetyComment implements Action {
```
Comment (mgoldson) [inline src/modules/tests/test-data/common/highway-code-safety/highway-code-safety.actions.ts]: For consistency with the names of other actions, this should probably be 'HighwayCodeSafetyAddComment'

File: src/modules/tests/test-data/common/highway-code-safety/highway-code-safety.actions.ts [hunk 5]
```diff
@@ -0,0 +1,34 @@
+import { Action } from '@ngrx/store';
+
+export const TOGGLE_HIGHWAYCODE_SAFETY = '[HighwaycodeSafety] Toggle Highwaycode Safety';
```
Comment (mgoldson) [inline src/modules/tests/test-data/common/highway-code-safety/highway-code-safety.actions.ts]: Should HIGHWAYCODE be separated by an underscore (two words)?

File: src/pages/test-report/test-report.analytics.effects.ts [hunk 6]
```diff
@@ -344,6 +345,52 @@ export class TestReportAnalyticsEffects {
     }),
   );
 
+  @Effect()
+highwayCodeSafetyAddDrivingFault$ = this.actions$.pipe(
```
Comment (szabikr) [inline src/pages/test-report/test-report.analytics.effects.ts]: This is going to be really weird, but can you indent these lines?

**Review by szabikr — APPROVED** [review]
For keeping the consistency it would be good if you made an indentation for the analytics effects before merging

**Review by mgoldson — APPROVED** [review]
Good work Ross!

**Pull request merged**